### PR TITLE
Regenerates R API docs & adds instructions for updating R documentation when making API changes

### PR DESCRIPTION
**Pull request by smurching**
Regenerates R API docs, and adds a bash script + instructions for regenerating R API docs to our contributor guide. There are currently no instructions for how to do this so it's pretty confusing for poeple trying to build the docs (i.e. I basically had to google around & figure out the approach).

### CODE REVIEW

File: docs/build-rdoc.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -ex
+pushd ../mlflow/R/mlflow
+Rscript -e 'install.packages("devtools", repos = "https://cloud.r-project.org")'
+Rscript -e 'devtools::install_deps(dependencies = TRUE)'
+rm -rf man
+Rscript -e "roxygen2::roxygenise()"
```
Comment (dbczumar) [inline docs/build-rdoc.sh]: Should we also install `roxygen2` here? 

```
The downloaded binary packages are in
        /var/folders/rq/g5vkbkv93r12yb1t374lfj0r0000gn/T//Rtmp0CIm6G/downloaded_packages
+ rm -rf man
+ Rscript -e 'roxygen2::roxygenise()'
Error in loadNamespace(name) : there is no package called ‘roxygen2’
Calls: :: ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
Execution halted
make: *** [html] Error 1
```

**Review by dbczumar — COMMENTED** [review]
@smurching Left a couple small comments. The most important is the one about the expected process for contributing w.r.t docs changes.

File: docs/build-rdoc.sh (hunk 1 above)
Comment (smurching) [reply docs/build-rdoc.sh]: Thanks for the catch, switched this to use `install_dev_deps` which should install `roxygen2`: https://www.rdocumentation.org/packages/devtools/versions/1.13.6/topics/install_deps

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -58,6 +58,9 @@ and `install <https://maven.apache.org/install.html>`_ Maven. You can then build
   cd mlflow/java
   mvn compile test
 
+If making API changes, please regenerate API documentation as described in `Writing Docs`_.
```
[the remark this answers is not in the document]
Comment (smurching) [reply CONTRIBUTING.rst]: Yep good question, I'll update this to be clearer.

I think in an ideal world we'd keep the docs in master up to date, as it makes it easier to update e.g. examples & guides in the docs ([like this one](https://mlflow.org/docs/latest/tracking.html#organizing-runs-in-experiments)) if we can link to up-to-date API docs. It's hard to force contributors to commit doc changes without a test, but actually wouldn't be too hard to add such a test in a follow-up (basically just regenerate the docs & verify they match what's in the user's PR). Having contributors commit doc changes should also help contributors & reviewers clearly see if there are any changes to public APIs, which could be nice. The main downside is that PRs may become larger, but hopefully not by much unless they're changing a bunch of APIs. Let me know what you think!

File: mlflow/R/mlflow/man-roxygen/roxlate-client.R [hunk 3]
```diff
@@ -1,3 +1,2 @@
 #' @param client (Optional) An `mlflow_client` object.
-#' @family tracking functions
```
Comment (smurching) [inline mlflow/R/mlflow/man-roxygen/roxlate-client.R]: Removed this family annotation, which causes the @seealso blobs to disappear from the R public API docs. The main downside is that if users build R docs on their own / view docs for an individual function like `mlflow_log_metric` e.g. in RStudio, there won't be cross-links to other tracking API docs. However this seems like a worthwhile tradeoff for having clean public docs.

File: docs/build-rdoc.sh (hunk 1 above)
Comment (dbczumar) [reply docs/build-rdoc.sh]:
```
+ Rscript document.R
Error in loadNamespace(name) : there is no package called ‘Rd2md’
Calls: :: ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
Execution halted
make: *** [html] Error 1
```

**Comment by smurching** [comment]
Alright...I tested out the doc build using the official R docker image - hopefully it now works for you @dbczumar :P, thanks for all the catches & sorry for the back and forth. There is one dependency (pandoc) you need to install manually to build R docs, since there's no platform-independent installer that I can put in the `build-rdoc.sh` bash script AFAICT - installation instructions for pandoc are included in the [contributor guide](https://github.com/mlflow/mlflow/pull/1124/files#diff-ea1f65a8eaac0cec7ae151dfdbbbc587R32). I also tried to split up the Java & R doc generation commands so that users don't need to install Java/Maven to build R docs & vice versa.

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -26,6 +28,12 @@ We recommend installing MLflow in its own conda environment for development, as
 You can verify that ``npm`` is on the PATH by running ``npm -v``, and
 `install npm <https://www.npmjs.com/get-npm>`_ if needed.
 
+If contributing to MLflow's R APIs, install `R <https://cloud.r-project.org/>`_. For changes to R
+documentation, also install `pandoc <https://pandoc.org/installing.html/>`_. If contributing
```
Comment (dbczumar) [inline CONTRIBUTING.rst]: This is an extreme nit, but I get a 404 when accessing `https://pandoc.org/installing.html/` with the trailing slash. `https://pandoc.org/installing.html` (no trailing slash) seems to work.

**Review by dbczumar — COMMENTED** [review]
@smurching The code looks good to me in general. However, when generating the docs, I get a lot of Sphinx errors related to the R APIs:
```
/Users/czumar/mlflow/docs/source/R-api.rst:180: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:180: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:202: WARNING: Inline literal start-string without end-string.
/Users/czumar/mlflow/docs/source/R-api.rst:510: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:890: WARNING: Inline literal start-string without end-string.
/Users/czumar/mlflow/docs/source/R-api.rst:1072: WARNING: Inline literal start-string without end-string.
/Users/czumar/mlflow/docs/source/R-api.rst:1095: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:1095: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:1095: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:1095: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:1189: WARNING: Inline literal start-string without end-string.
/Users/czumar/mlflow/docs/source/R-api.rst:1224: WARNING: Inline literal start-string without end-string.
/Users/czumar/mlflow/docs/source/R-api.rst:1231: WARNING: Inline literal start-string without end-string.
/Users/czumar/mlflow/docs/source/R-api.rst:1235: WARNING: Inline literal start-string without end-string.
/Users/czumar/mlflow/docs/source/R-api.rst:1255: WARNING: Inline literal start-string without end-string.
/Users/czumar/mlflow/docs/source/R-api.rst:1691: WARNING: Inline literal start-string without end-string.
/Users/czumar/mlflow/docs/source/R-api.rst:1711: WARNING: Inline literal start-string without end-string.
/Users/czumar/mlflow/docs/source/R-api.rst:1758: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:1758: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:1796: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:1796: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:1796: WARNING: Unknown interpreted text role "raw-latex".
/Users/czumar/mlflow/docs/source/R-api.rst:1818: WARNING: Unknown interpreted text role "raw-latex".
```

Additionally, the R docs don't seem to render very well:

Screen Shot 2019-05-09 at 10 23 43 PM

Screen Shot 2019-05-09 at 10 25 47 PM

Can you reproduce this? Is it an issue with our docstrings?

**Review by dbczumar — APPROVED** [review]
LGTM, pending the following updates:

1. Update the contribution guide to specify the version of `pandoc` that should be used for generating R documentation (this fixes the doc generation issues described in my previous comment)

2. Remove the trailing slash from `https://pandoc.org/installing.html/` in the contributing guide.

**Pull request merged**